code/provision/playbooks/roles/rke-kubernetes/tasks/main.yml

```diff
@@ -0,0 +1,58 @@
+---
+- name: Download RKE binary
+  get_url:
+    url: "{{ rke_binary_url }}"
+    checksum: "{{ rke_binary_checksum }}"
+    dest: /usr/local/bin/rke
+    mode: "+rx"
+  register: rke_binary
+  delegate_to: 127.0.0.1
+
+- name: Ensure config directory exists
+  file:
+    path: "{{ rke_config_dir }}"
+    state: directory
+  delegate_to: 127.0.0.1
+
+- name: Create/update RKE config file
+  template:
+    src: rke_cluster.yml
+    dest: "{{ rke_config_dir }}/cluster.yml"
+    mode: u=rw,g=r,o=
+  register: rke_cluster_config
+  delegate_to: 127.0.0.1
+
+- name: Run RKE installer
+  shell: |
+    set -eo pipefail
+    rke up | tee /var/log/rke-up.log
+  args:
+    chdir: "{{ rke_config_dir }}"
+  when: rke_binaryis changed or rke_cluster_config is changed or rke_force_configure
+  delegate_to: 127.0.0.1
+
+- name: Install kubectl
+  get_url:
+    url: https://storage.googleapis.com/kubernetes-release/release/{{ rke_kubernetes_version.split('-', 1)[0] }}/bin/linux/amd64/kubectl
+    dest: /usr/local/bin/kubectl
+    mode: "+rx"
+  delegate_to: 127.0.0.1
+
+- name: Ensure $HOME/.kube exists
+  file:
+    path: "{{ ansible_env.HOME }}/.kube"
+    state: directory
+  delegate_to: 127.0.0.1
+
+- name: Copy kubeconfig to default location
+  copy:
+    remote_src: yes
+    src: "{{ rke_config_dir }}/kube_config_cluster.yml"
+    dest: "{{ ansible_env.HOME }}/.kube/config"
+  delegate_to: 127.0.0.1
+
+- name: Install jq
+  apt:
+    update_cache: yes
+    name: jq=1.5+dfsg-2
```

Comment:
Not sure we need to pin a version of jq?

Reply:
Not sure we need to as well. I have just gone with what we have currently - https://github.com/NERC-CEH/datalab/blob/master/code/provision/playbooks/roles/kubernetes-common/defaults/main.yml